site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+    if (call.selected_field !== 'tags') {
+        const callResponse: AppCallResponse = {
+            type: 'error',
+            error: `Unexpected lookup field: ${call.selected_field}`,
+        };
+
+        res.json(callResponse);
+        return;
+    }
+
+    // TicketTag represents a tag in the external ticket system
+    let tags: TicketTag[];
+    try {
+        // Use some other service to process the autocomplete request
+        tags = await autocompleteTags(call.query);
+    } catch (e) {
+        const callResponse: AppCallResponse = {
+            type: 'error',
+            error: `Error performing autocomplete: ${e.message}`,
+        };
+
+        res.json(callResponse);
+        return;
+    }
+
+    const selectOptions: AppSelectOption[] = tags.map(tag => ({
+        label: tag.name,
+        value: tag.id,
+    }));
+
+    const callResponse: AppCallResponse = {
+        type: 'ok',
+        data: {
+            items: selectOptions,
+        },
+    };
+
+    res.json(callResponse);
+});
+```
+
+Now we have our form integrated with an external system's autocomplete service, with just a few lines of code. Note that we also have access to all of the currently selected values in the form whenever we receive a `lookup` request. For example, we have access to the current Summary and Description through `call.values.summary` and `call.values.description`.
+
+### Redefining the form after user actions
+
+Our use case may be more complex, and the form's structure needs to be updated as the user fills out certain fields. For example, let's say the user needs to choose an Issue Type, which should then cause the form to show other form fields based on the Issue Type chosen by the user. We'll need to use the "form refresh" feature of the App forms to accomplish this.
+
+Let's add the "Issue Type" field to the form:
+
+```ts
+const issueTypeField: AppField = {
+    name: 'issue_type',
+    modal_label: 'Issue Type',
+    type: 'static_select',
+    value: null,
+    refresh: true, // This tells the frontend to contact our App when the user has chosen a value for this field. We then have the opportunity to redefine a new form with access to the current form's state.
+    options: [
+        {label: 'Story', value: 'story'},
+        {label: 'Task', value: 'task'},
+        {label: 'Bug', value: 'bug'},
+    ],
+}
+```
+
+When the Issue Type value is selected, our App will receive a request at the endpoint `/forms/create-ticket/form`. The request body will be similar to that of a form submission, but it will also include what field was selected in the `call.selected_field` property.
+
+Note that we can return an entirely new form here. We can also create a flow between several forms by responding to a form submission with a new form.
```

Comment:
This is confusing right here. I would remove it from here and create a new section somewhere else.